dataRetriever/resolvers/headerResolver.go

```diff
@@ -123,6 +136,43 @@ func (hdrRes *HeaderResolver) resolveHeaderFromNonce(key []byte) ([]byte, error)
 	return buff, nil
 }
 
+// resolveHeaderFromHash resolves a header using its key (header hash)
+func (hdrRes *HeaderResolver) resolveHeaderFromHash(key []byte) ([]byte, error) {
+	value, ok := hdrRes.headers.Peek(key)
+	if !ok {
+		return hdrRes.hdrStorage.Get(key)
```

Comment:
if !ok we can just return notFound, why do we need to do the get? the only thing get does in addition is updating the recently used-ness of the element.

Reply:
Because the peek is done on the data pool and the get is done on the storage